File: binds/java.v1/src/service.java
import com.karalabe.iris.Connection;
import com.karalabe.iris.Service;
import com.karalabe.iris.ServiceHandler;
import com.karalabe.iris.Tunnel;
import com.karalabe.iris.exceptions.InitializationException;
import com.karalabe.iris.exceptions.RemoteException;

import java.io.IOException;

// START OMIT
public class ServiceDemo implements ServiceHandler {
    // Handler initialization, invoked after successful registration
    @Override public void init(Connection connection) throws InitializationException { } // HLreg

    public static void main(String args[]) {
        // Register a micro-service instance into the Iris network
        try (Service service = new Service(55555, "Javatar service", new ServiceDemo())) { // HLreg
            // Do something, before tearing down
        }
        catch (IOException | InitializationException | InterruptedException e) {
            throw new RuntimeException("Failed to register the service", e);
        }
    }

    // Remaining callbacks methods, not used in this demo
    @Override public void handleBroadcast(byte[] message)                                { }
    @Override public byte[] handleRequest(byte[] req) throws RemoteException { return req; }
    @Override public void handleTunnel(Tunnel tunnel)                                    { }
    @Override public void handleDrop(Exception reason)                                   { }
}
// END OMIT

File: binds/scala.v1/src/service.scala
import com.karalabe.iris.{Service, Connection, ServiceHandler, Tunnel}

// START OMIT
class ServiceDemo extends ServiceHandler {
    // Handler initialization, invoked after successful registration
    override def init(connection: Connection) { } // HLreg

    // Remaining callbacks methods, not used in this demo
    override def handleBroadcast(message: Array[Byte])              { }
    override def handleRequest(request: Array[Byte]): Array[Byte] = {return request }
    override def handleTunnel(tunnel: Tunnel)                       { }
    override def handleDrop(reason: Exception)                      { }
}

object ServiceDemo {
    def main(args: Array[String]) {
        // Register a micro-service instance into the network
        val service = new Service(55555, "Lausanne service", new ServiceDemo()) // HLreg

        // Unregister the service
        service.close() // HLreg
    }
}
// END OMIT
